CIHM-Meta/lib/CIHM/Meta/Hammer2/Process.pm

```diff
@@ -773,4 +697,598 @@ sub getAccessCollections {
     }
 }
 
+sub deletedocument {
+    my ($self) = @_;
+
+    $self->update_couch( $self->cosearch2db );
+    $self->update_couch( $self->copresentation2db );
+}
+
+sub adddocument {
+    my ($self) = @_;
+
+    $self->process_attachment();
+
+    # Map also counts for a minimum of repos, so adding in current array
+    # to presentation.
+    $self->presentdoc->{ $self->slug }->{'repos'} = $self->docdata->{'repos'}
+      if defined $self->docdata->{'repos'};
+
+    # All Items should have a date
+    $self->presentdoc->{ $self->slug }->{'updated'} =
+      DateTime->now()->iso8601() . 'Z';
+
+    # Note: Not stored within pages, so no need to loop through all keys
+    my @collections = sort keys %{ $self->{collections} };
+    $self->presentdoc->{ $self->slug }->{'collection'} = \@collections;
+    $self->searchdoc->{ $self->slug }->{'collection'}  = \@collections;
+
+    # New key to build better breadcrumbs in the future
+    $self->presentdoc->{ $self->slug }->{'collection_tree'} =
```

Comment:
This collection_tree is a new field that can hopefully be used in the future to build a better breadcrumb tree.  The concept of "series" and "issues" needs to be generalised better for a IIIF context.

Collections aren't clean at the moment.  Manifests are regularly added directly to collections which sub-collections are also a member of.

See  https://access.canadiana.ca/object/edit/69429/m0ns0ks6mp0t as an example:

It should only be in "oocihm.8_06185: Gazette des campagnes", and not also in "eco" and "per"

oocihm.8_06185 at https://access.canadiana.ca/object/edit/69429/s0zc7rn31g02 should only be in "per" (as "per" is already in "eco")

This is all post-PASplit data cleanup that wasn't possible to do until that project was completed.

Reply:
@brittnylapierre ,

I created https://github.com/crkn-rcdr/cap/issues/138 to discuss the legacy "issue" related problem in more detail.

The `collection_tree` field I'm adding to "presentation" documents should be understood as an **experiment** to push the conversations forward. It is expected that we may change it as we solidify our plans.  I you like the idea I've come up with we can document, but we haven't yet created a place in our documentation for cosearch and copresentation #19 .



Field is an array of immediate ancestors (collections this document is a member of). An empty array means there are no immediate ancestors. Elements of array are a structure with:


  * label (IIIF string, so with language/etc)
  * noid
  * slug
  * behaviour
  * collections, which is an array of immediate ancestors (collections this collection is a member of). Array is empty if there are no immediate ancestors.

Using the oocihm.8_06185_271 from above as an example (which shows why we don't want to have so much duplication in the tree).



```
"collection_tree": [
    {
      "label": {
        "fr": "Notre mémoire en ligne",
        "en": "Early Canadiana Online"
      },
      "behavior": "unordered",
      "collections": [],
      "slug": "eco",
      "noid": "69429/s0df6k06z33x"
    },
    {
      "label": {
        "fr": "Anciens périodiques canadiens",
        "en": "Early Canadian Periodicals"
      },
      "noid": "69429/s0kk94747v7d",
      "slug": "per",
      "collections": [
        {
          "label": {
            "fr": "Notre mémoire en ligne",
            "en": "Early Canadiana Online"
          },
          "behavior": "unordered",
          "collections": [],
          "slug": "eco",
          "noid": "69429/s0df6k06z33x"
        },
        {
          "label": {
            "none": "serials"
          },
          "slug": "serials",
          "noid": "69429/s0hm52f7kt36",
          "behavior": "unordered",
          "collections": [
            {
              "label": {
                "fr": "Canadiana en ligne",
                "en": "Canadiana Online"
              },
              "noid": "69429/s0j678s4kv36",
              "slug": "online",
              "collections": [],
              "behavior": "unordered"
            }
          ]
        }
      ],
      "behavior": "unordered"
    },
    {
      "noid": "69429/s0zc7rn31g02",
      "slug": "oocihm.8_06185",
      "collections": [
        {
          "collections": [],
          "behavior": "unordered",
          "noid": "69429/s0df6k06z33x",
          "slug": "eco",
          "label": {
            "fr": "Notre mémoire en ligne",
            "en": "Early Canadiana Online"
          }
        },
        {
          "label": {
            "en": "Early Canadian Periodicals",
            "fr": "Anciens périodiques canadiens"
          },
          "behavior": "unordered",
          "collections": [
            {
              "label": {
                "en": "Early Canadiana Online",
                "fr": "Notre mémoire en ligne"
              },
              "collections": [],
              "behavior": "unordered",
              "noid": "69429/s0df6k06z33x",
              "slug": "eco"
            },
            {
              "collections": [
                {
                  "label": {
                    "en": "Canadiana Online",
                    "fr": "Canadiana en ligne"
                  },
                  "collections": [],
                  "behavior": "unordered",
                  "noid": "69429/s0j678s4kv36",
                  "slug": "online"
                }
              ],
              "behavior": "unordered",
              "noid": "69429/s0hm52f7kt36",
              "slug": "serials",
              "label": {
                "none": "serials"
              }
            }
          ],
          "slug": "per",
          "noid": "69429/s0kk94747v7d"
        }
      ],
      "behavior": "multi-part",
      "label": {
        "none": "Gazette des campagnes"
      }
    }
  ],
```